cloud/kubernetes/yugabyte-statefulset-rf-1.yaml

```diff
@@ -0,0 +1,256 @@
+  replicas: 1
+  selector:
+    matchLabels:
+      app: yb-master
+  template:
+    metadata:
+      labels:
+        app: yb-master
+    spec:
+      affinity:
+        # Set the anti-affinity selector scope to YB masters.
+        podAntiAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+          - weight: 100
+            podAffinityTerm:
+              labelSelector:
+                matchExpressions:
+                - key: app
+                  operator: In
+                  values:
+                  - yb-master
+              topologyKey: kubernetes.io/hostname
+      containers:
+      - name: yb-master
+        image: yugabytedb/yugabyte:latest
+        imagePullPolicy: Always
+        env:
+        - name: GET_HOSTS_FROM
+          value: dns
+        - name: POD_IP
+          valueFrom:
+            fieldRef:
+              fieldPath: status.podIP
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        command:
+          - "/home/yugabyte/bin/yb-master"
+          - "--fs_data_dirs=/mnt/data0"
+          - "--rpc_bind_addresses=$(POD_IP):7100"
+          - "--server_broadcast_addresses=$(POD_NAME).yb-masters:7100"
+          - "--use_private_ip=never"
+          - "--master_addresses=yb-masters.default.svc.cluster.local:7100"
+          - "--master_replication_factor=1"
+          - "--replication_factor=1"
+          - "--logtostderr"
+        ports:
+        - containerPort: 7000
+          name: master-ui
+        - containerPort: 7100
+          name: master-rpc
+#        readinessProbe:
+#          tcpSocket:
+#            port: 7100
+#          initialDelaySeconds: 5
+#          periodSeconds: 10
+#        livenessProbe:
+#          tcpSocket:
+#            port: 7100
+#          initialDelaySeconds: 15
+#          periodSeconds: 10
+
+        # These volume mounts are persistent. They are like inline claims,
+        # but not exactly because the names need to match exactly one of
+        # the stateful pod volumes.
+        volumeMounts:
+        - name: datadir
+          mountPath: /mnt/data0
+  updateStrategy:
+    type: RollingUpdate
+  # These are converted to volume claims by the controller
+  # and mounted at the paths mentioned above.
+  # do not use these in production until ssd GCEPersistentDisk or other ssd pd
+  volumeClaimTemplates:
+  - metadata:
+      name: datadir
+    spec:
+      accessModes: [ "ReadWriteOnce" ]
+      resources:
+        requests:
+          storage: 10Gi
+      storageClassName: standard
+
+---
+apiVersion: v1
+kind: Service
+metadata:
+  # This is a "headless" service for the yb-tserver which exists to allow discovery of the set of
+  # member pods (tservers).
+  name: yb-tservers
+  labels:
+    app: yb-tserver
+spec:
+  clusterIP: None
+  ports:
+  - name: ui
+    port: 9000
+  - name: rpc-port
+    port: 9100
+  - name: cassandra
+    port: 9042
+  - name: redis
+    port: 6379
```

Comment:
Having the ysql headless service would be good as well, though its not directly used for anything it helps discovery.